### ISSUE TITLE
Support new inline TypedDict form

### ISSUE THREAD

**Issue by kaste** [astral-sh/ruff#22619]
### Summary

The mypy Inline TypedDict syntax is not supported  https://mypy.readthedocs.io/en/latest/typed_dict.html#inline-typeddict-types

https://play.ruff.rs/9dd9008b-3148-42b6-bba6-b9eff0f5a5e2

```py
from typing import TypeAlias

def test_values() -> {"width": int, "description": str}:
    return {"width": 42, "description": "test"}

Y: TypeAlias = {"a": int, "b": int}
type Z = {"a": int, "b": int}
```
```
Diagnostics (7)
Undefined name `width` (F821) [Ln 3, Col 24]
Undefined name `description` (F821) [Ln 3, Col 38]
Undefined name `a` (F821) [Ln 6, Col 18]
Undefined name `b` (F821) [Ln 6, Col 28]
Cannot use `type` alias statement on Python 3.10 (syntax was added in Python 3.12) (invalid-syntax) [Ln 7, Col 1]
Undefined name `a` (F821) [Ln 7, Col 12]
Undefined name `b` (F821) [Ln 7, Col 22]
```

### Version

v0.14.13

**Comment by ntBre** [comment]
Thanks for the report! Based on the linked docs, this is an "experimental (non-standard) feature" feature even in mypy, so I think we may want to wait and see if this becomes more widely supported. It looks like [ty](https://play.ty.dev/2c062c5c-8a77-4cce-ac02-7d71adf3e316) doesn't currently support this either.

**Comment by Kilo59** [comment]
@ntBre 
Would you accept a PR that adds this support, even if only by opting-in to some config setting to not flag F821 on this?

I don't actually know how difficult this would be as we would want to ensure we don't regress on the non inline typeddicts.
